src/main/java/net/mcreator/workspace/Workspace.java

```diff
@@ -294,6 +294,21 @@ public long getMCreatorVersion() {
 		fileManager.close();
 	}
 
+	@Override public boolean equals(Object o) {
+		if (this == o)
+			return true;
+
+		if (o instanceof Workspace other) {
+			return getWorkspaceFolder().equals(other.getWorkspaceFolder());

```

Comment:
Not a wise approach to compare files. See https://stackoverflow.com/questions/8930859/java-file-equals

equals can return true potining to same file due to different absolute path.

Files.isSameFile may be the answer here